core/state_transition.go

```diff
@@ -467,12 +470,16 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		fee := new(uint256.Int).SetUint64(st.gasUsed())
 		fee.Mul(fee, effectiveTipU256)
 		st.state.AddBalance(st.evm.Context.Coinbase, fee)
-		// CHANGE(taiko): basefee is not burnt, but sent to a treasury instead.
+		// CHANGE(taiko): basefee is not burnt, but sent to a treasury and block.coinbase instead.
 		if st.evm.ChainConfig().Taiko && st.evm.Context.BaseFee != nil && !st.msg.IsAnchor {
-			st.state.AddBalance(
-				st.getTreasuryAddress(),
-				uint256.MustFromBig(new(big.Int).Mul(st.evm.Context.BaseFee, new(big.Int).SetUint64(st.gasUsed()))),
+			totalFee := new(big.Int).Mul(st.evm.Context.BaseFee, new(big.Int).SetUint64(st.gasUsed()))
+			feeCoinbase := new(big.Int).Div(
```

Comment:
Here make sure `BasefeeSharingPctg` is less than 100.

Reply:
The smart contract should make sure of that?

Reply:
We'd better do a double check here to make sure nothing goes wrong.